modules/auxiliary/scanner/mysql/mysql_login.rb

```diff
@@ -211,6 +211,10 @@ def session_setup(result, client)
       'PASSWORD'      => result.credential.private
     }
 
+    server_vars = my_session.client.query_server_vars
```

Comment:
I don't like the thought of folk accidentally creating new MySQL sessions that don't have a arch/platform present; Could we maybe update the MySQL's initialize method to require platform/arch in the opts to ensure that the client provides these values

```ruby
    my_session = Msf::Sessions::MySQL.new(rstream, { client: client, *client.get_platform_and_arch })
```

And (spitballed) the initialize method:

```ruby
def initialize(rstream, opts)
  self.arch = opts.fetch(:arch)
  self.platform = opts.fetch(:platform)

```